doc/languages-frameworks/maven.section.md

```diff
@@ -0,0 +1,355 @@
+
+### CLASSPATH
+
+> This is ideal if you are providing a derivation for _nixpkgs_ and don't want to patch the project's _pom.xml_.
+
+We will read the Maven repository and flatten it to a single list. This list will then be concatenated with the _CLASSPATH_ separator to create the full classpath.
+
+We make sure to provide this classpath to the _makeWrapper_.
+
+
+```nix
+{ stdenv, lib, maven, callPackage, makeWrapper, jre }:
+let
+  repository = callPackage ./build-maven-repository.nix { };
+in stdenv.mkDerivation rec {
+  pname = "maven-demo";
+  version = "1.0";
+
+  src = builtins.fetchTarball
+    "https://github.com/fzakaria/nixos-maven-example/archive/main.tar.gz";
+  buildInputs = [ maven makeWrapper ];
+
+  buildPhase = ''
+    echo "Using repository ${repository}"
+    mvn --offline -Dmaven.repo.local=${repository} package;
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+
+    classpath=$(find ${repository} -name "*.jar" -printf ':%h/%f');
+    install -Dm644 target/${pname}-${version}.jar $out/share/java
+    # create a wrapper that will automatically set the classpath
+    # this should be the paths from the dependency derivation
+    makeWrapper ${jre}/bin/java $out/bin/${pname} \
+          --add-flags "-classpath $out/share/java/${pname}-${version}.jar:''${classpath#:}" \
+          --add-flags "Main"
+  '';
+}
+```
+
+### MANIFEST file via Maven Plugin
+
+> This is ideal if you are the project owner and want to change your _pom.xml_ to set the CLASSPATH within it.
+
+Augment the _pom.xml_ to create a JAR with the following manifest:
+```xml
+<build>
+  <plugins>
+    <plugin>
+        <artifactId>maven-jar-plugin</artifactId>
+        <configuration>
+            <archive>
+                <manifest>
+                    <addClasspath>true</addClasspath>
+                    <classpathPrefix>../../repository/</classpathPrefix>
+                    <classpathLayoutType>repository</classpathLayoutType>
+                    <mainClass>Main</mainClass>
+                </manifest>
+                <manifestEntries>
+                    <Class-Path>.</Class-Path>
+                </manifestEntries>
+            </archive>
+        </configuration>
+    </plugin>
+  </plugins>
+</build>
+```
+
+The above plugin instructs the JAR to look for the necessary dependencies in the _lib/_ relative folder. The layout of the folder is also in the _maven repository_ style.
+
+```bash
+❯ unzip -q -c $(nix-build --no-out-link runnable-jar.nix)/share/java/maven-demo-1.0.jar META-INF/MANIFEST.MF
+
+Manifest-Version: 1.0
+Archiver-Version: Plexus Archiver
+Built-By: nixbld
+Class-Path: . ../../repository/com/vdurmont/emoji-java/5.1.1/emoji-jav
+ a-5.1.1.jar ../../repository/org/json/json/20170516/json-20170516.jar
+Created-By: Apache Maven 3.6.3
+Build-Jdk: 1.8.0_265
+Main-Class: Main
+```
+
+We will modify the derivation above to add a symlink to our repository so that it's accessible to our JAR during the _installPhase_.
```

Comment:
```suggestion
We will modify the derivation above to add a symlink to our repository so that it's accessible to our JAR during the `installPhase`.
```